src/azure-cli/azure/cli/command_modules/acs/agentpool_decorator.py

```diff
@@ -2228,5 +2243,7 @@ def update_agentpool(self, agentpool: AgentPool) -> AgentPool:
             self.context.get_cluster_name(),
             self.context.get_nodepool_name(),
             agentpool,
+            if_match=self.context.get_if_match(),
+            if_none_match=self.context.get_if_none_match(),
```

Comment:
Test based on the old profile (using API version 2020-11-01) failed with the following error.

https://dev.azure.com/azclitools/public/_build/results?buildId=208123&view=logs&j=4e4d232d-230a-55c9-0730-4c1a9c36b9e5&t=6d1bdb99-245b-5117-da18-01d8df6f72c0&l=1836
> >           response = self.session.request(  # type: ignore
>                 request.method,
>                 request.url,
>                 headers=request.headers,
>                 data=request.data,
>                 files=request.files,
>                 verify=kwargs.pop("connection_verify", self.connection_config.verify),
>                 timeout=timeout,
>                 cert=kwargs.pop("connection_cert", self.connection_config.cert),
>                 allow_redirects=False,
>                 proxies=proxies,
>                 **kwargs
>             )
> E           TypeError: request() got an unexpected keyword argument 'if_match'

something like this can be used as a workaround to support both the old profile with 2020-11-01 as the default API version and the latest profile with the latest API version.
Please move `from azure.cli.core.cloud import get_active_cloud` to the top of this file
```python
        from azure.cli.core.cloud import get_active_cloud
        active_cloud = get_active_cloud(self.cmd.cli_ctx)
        if active_cloud.profile != "latest":
            return sdk_no_wait(
                self.context.get_no_wait(),
                self.client.begin_create_or_update,
                self.context.get_resource_group_name(),
                self.context.get_cluster_name(),
                self.context.get_nodepool_name(),
                agentpool,
                headers=self.context.get_aks_custom_headers(),
            )
        else:
            return sdk_no_wait(
                self.context.get_no_wait(),
                self.client.begin_create_or_update,
                self.context.get_resource_group_name(),
                self.context.get_cluster_name(),
                # validated in "init_agentpool", skip to avoid duplicate api calls
                self.context._get_nodepool_name(enable_validation=False),
                agentpool,
                if_match=self.context.get_if_match(),
                if_none_match=self.context.get_if_none_match(),
                headers=self.context.get_aks_custom_headers(),
            )
```

Reply:
Updated